### DISCUSSION TITLE
Identify the fiber where the code is running

### DISCUSSION THREAD

**Discussion by matejthetree** [reactphp/reactphp#562] [Q&A]
I have heavy code with bunch of fibers spawning each second with async. It would be helpful if I could write a middleware to logger that helps to identify the current fiber where the code is running. 

This would help to filter out the logs for each request I get with socket and to follow the code for that particular instance.

can this be done?

**Comment by WyriHaximus** [comment]
It isn't on our official debated and agreed up on roadmap for `async`. But it is on my personal list to look into as such functionality will help with tracing. Will specifically look at adding MDC (Mapped Diagnostic Context) functionality to either async or proof it first in a stand alone package. In theory it is pretty simple to build, but both as a stand alone package and build in have their tradeoffs. So I want to explore those thoroughly before discussing it with @SimonFrings and @clue.

**Comment by SimonFrings** [comment]
@matejthetree You could also use `Fiber::getCurrent` to get the currently executing Fiber instance. I also answered a similar ticket in #546, so maybe this also helps as well :+1:
